pandas/tests/scalar/timestamp/test_timezones.py

```diff
@@ -21,17 +21,48 @@ class TestTimestampTZOperations:
     # Timestamp.tz_localize
 
     def test_tz_localize_pushes_out_of_bounds(self):
-        msg = "^$"
         # GH#12677
         # tz_localize that pushes away from the boundary is OK
         pac = Timestamp.min.tz_localize("US/Pacific")
+        pac_timestamp_data = {
+            "year": pac.year,
+            "month": pac.month,
+            "day": pac.day,
+            "hour": pac.hour,
+            "min": pac.minute,
+            "sec": pac.second,
+            "us": pac.microsecond,
+            "ps": 0,
+            "as": 0,
+        }
+        msg = (
+            "Timestamp {} cannot be converted within implementation "
+            "bounds {} and {}".format(pac_timestamp_data, Timestamp.min, Timestamp.max)
+        )
```

Comment:
the user-facing message shown is ```Timestamp {'year': 1677, 'month': 9, 'day': 21, 'hour': 0, 'min': 12, 'sec': 43, 'us': 145225, 'ps': 0, 'as': 0} cannot be converted within implementation bounds 1677-09-21 00:12:43.145225 and 2262-04-11 23:47:16.854775807```

looking for simplification here while still being helpful see https://github.com/pandas-dev/pandas/pull/32979/files#r399153608